kedro/framework/project/__init__.py

```diff
@@ -259,3 +261,58 @@ def validate_settings():
         )
 
     importlib.import_module(f"{PACKAGE_NAME}.settings")
+
+
+def find_pipelines() -> Dict[str, Pipeline]:
+    """Automatically find modular pipelines having a ``create_pipeline``
+    function. By default, projects created using Kedro 0.18.3 and higher
+    call this function to autoregister pipelines upon creation/addition.
+
+    Returns:
+        A generated mapping from pipeline names to ``Pipeline`` objects.
+
+    Warns:
+        UserWarning: When a module does not expose a ``create_pipeline``
+            function, the ``create_pipeline`` function does not return a
+            ``Pipeline`` object, or if the module import fails up front.
+    """
+    pipelines_dict = {"__default__": pipeline([])}
+    for pipeline_dir in importlib_resources.files(
+        f"{PACKAGE_NAME}.pipelines"
+    ).iterdir():
+        if not pipeline_dir.is_dir():
+            continue
+        pipeline_name = pipeline_dir.name
+        if pipeline_name == "__pycache__":
+            continue
+        try:
+            pipeline_module = importlib.import_module(
+                f"{PACKAGE_NAME}.pipelines.{pipeline_name}"
```

Comment:
Just to check: you need to reconstruct the dotted module path here and can't just do something like `import_module(pipeline_dir)`?

Reply:
Correct; `importlib.import_module` takes a dotted path, as in an import statement, not a directory.